### PR TITLE
HRMP Channel Establishment with System chains

### PR DESCRIPTION
**Pull request by DrW3RK**
Opening channels with system chains https://paritytech.github.io/polkadot-sdk/master/polkadot_runtime_parachains/hrmp/pallet/dispatchables/fn.establish_channel_with_system.html

#6085

### CODE REVIEW

File: docs/build/build-hrmp-channels.md [hunk 1]
```diff
@@ -59,40 +59,21 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
 
 ## Opening HRMP Channels with System Parachains
 
-HRMP channel management involving system parachains takes place entirely on the Relay Chain. No
-action is required from the parachain origin.
+[`establish_channel_with_system`](https://paritytech.github.io/polkadot-sdk/master/polkadot_runtime_parachains/hrmp/pallet/dispatchables/fn.establish_channel_with_system.html)
```
Comment (joepetrowski) [inline docs/build/build-hrmp-channels.md]:
```suggestion
The [`establish_channel_with_system`](https://paritytech.github.io/polkadot-sdk/master/polkadot_runtime_parachains/hrmp/pallet/dispatchables/fn.establish_channel_with_system.html)
```

File: docs/build/build-hrmp-channels.md [hunk 2]
```diff
@@ -59,40 +59,21 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
+call can be used to establish a bi-directional channel between any parachain and a system chain.
+This call needs to be dispatched from the parachain as an XCM message that needs to get executed on
+the relay chain.
```
Comment (joepetrowski) [inline docs/build/build-hrmp-channels.md]:
```suggestion
This call needs to be dispatched from the parachain via an XCM program to execute on the Relay Chain.
```

File: docs/build/build-hrmp-channels.md [hunk 3]
```diff
@@ -59,40 +59,21 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
 
-Opening an HRMP channel with a system parachain requires an
-[OpenGov referendum](../learn/learn-guides-polkadot-opengov.md) using the
-[General Admin Track](../learn/learn-polkadot-opengov-origins.md#general-admin).
+For instance, the call to be executed on the relay chain to establish bi-directional channel between
+a parachain with `ParaID` 4428 and the Asset Hub with `ParaID` 1000 is `0x3c0ae8030000`
```
Comment (joepetrowski) [inline docs/build/build-hrmp-channels.md]: Does the para ID matter in this call?

File: docs/build/build-hrmp-channels.md [hunk 4]
```diff
@@ -59,40 +59,21 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
 
-Proposals should generally be a `batchAll` call containing two `forceOpenHrmpChannel` calls (one for
-each direction of the channel).
+![asset-hub-hrmp-channel](../assets/asset-hub/establish-channel-asset-hub.png)
 
-As an example, see [Referendum 280](https://polkadot.polkassembly.io/referenda/280), which opened
-two bi-directional channels with Asset Hub (one for Zeitgeist and one for Composable).
+Here is an example call with XCM message sent from the parachain with `ParaID` 4428 to Rococo relay
```
Comment (joepetrowski) [inline docs/build/build-hrmp-channels.md]: Same

File: docs/build/build-hrmp-channels.md (hunk 3 above)
Comment (profradha) [reply docs/build/build-hrmp-channels.md]: Not necessary. Removed it.

File: docs/build/build-hrmp-channels.md [hunk 5]
```diff
@@ -59,40 +59,20 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
+For instance, the call to be executed on the relay chain to establish bi-directional channel between
+a parachain and the Asset Hub with `ParaID` 1000 is `0x3c0ae8030000`
 
-Proposals should generally be a `batchAll` call containing two `forceOpenHrmpChannel` calls (one for
-each direction of the channel).
+![asset-hub-hrmp-channel](../assets/asset-hub/establish-channel-asset-hub.png)
 
-As an example, see [Referendum 280](https://polkadot.polkassembly.io/referenda/280), which opened
-two bi-directional channels with Asset Hub (one for Zeitgeist and one for Composable).
+Here is an example call with XCM message sent from the parachain to Rococo relay
```
Comment (CrackTheCode016) [inline docs/build/build-hrmp-channels.md]:
```suggestion
Here is an example call with an XCM program sent from the parachain to Rococo relay
```

File: docs/build/build-hrmp-channels.md [hunk 6]
```diff
@@ -59,40 +59,20 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
-HRMP channel management involving system parachains takes place entirely on the Relay Chain. No
-action is required from the parachain origin.
+The [`establish_channel_with_system`](https://paritytech.github.io/polkadot-sdk/master/polkadot_runtime_parachains/hrmp/pallet/dispatchables/fn.establish_channel_with_system.html)
+call can be used to establish a bi-directional channel between any parachain and a system chain.
+This call needs to be dispatched from the parachain via an XCM program to execute on the Relay Chain.
 
-Opening an HRMP channel with a system parachain requires an
-[OpenGov referendum](../learn/learn-guides-polkadot-opengov.md) using the
-[General Admin Track](../learn/learn-polkadot-opengov-origins.md#general-admin).
+For instance, the call to be executed on the relay chain to establish bi-directional channel between
```
Comment (CrackTheCode016) [inline docs/build/build-hrmp-channels.md]:
```suggestion
For instance, the call to be executed on the relay chain to establish a bi-directional channel between
```

File: docs/build/build-hrmp-channels.md [hunk 7]
```diff
@@ -59,40 +59,20 @@ of the `XcmExecutor`, which handles notifications of channel requests and accept
...
 
-Proposals should generally be a `batchAll` call containing two `forceOpenHrmpChannel` calls (one for
-each direction of the channel).
+![asset-hub-hrmp-channel](../assets/asset-hub/establish-channel-asset-hub.png)
 
-As an example, see [Referendum 280](https://polkadot.polkassembly.io/referenda/280), which opened
-two bi-directional channels with Asset Hub (one for Zeitgeist and one for Composable).
+Here is an example call with XCM message sent from the parachain to Rococo relay
+chain to establish channel a bi-directional channel with Rococo Asset Hub -
+`0x0f001f000301000314000400000000070010a5d4e81300000000070010a5d4e80006000300c16678419c183c0ae8030000140d01000001003145`
```
Comment (CrackTheCode016) [inline docs/build/build-hrmp-channels.md:73]: Not sure when I try to decode in PJS, it throws an error: 

Screenshot 2024-08-05 at 12 10 49 PM

**Review by CrackTheCode016 — COMMENTED** [review]
LGTM, just can't decode payload but maybe something from my end

File: docs/build/build-hrmp-channels.md (hunk 5 above)
Comment (joepetrowski) [reply docs/build/build-hrmp-channels.md]: I prefer "XCM program"

File: docs/build/build-hrmp-channels.md (hunk 7 above)
Comment (profradha) [reply docs/build/build-hrmp-channels.md:73]: Where have you tried to decode this call? This should be sent from the parachain (in our context, Educhain).

Comment (profradha) [reply docs/build/build-hrmp-channels.md:73]: image

A d was flipped to 0 in the call you copied to PJS UI. Not sure how.

Comment (profradha) [reply docs/build/build-hrmp-channels.md:73]: `0x0f001f000301000314000400000000070010a5d4e81300000000070010a5d4e80006000300c16678419c183c0ae8030000140d01000001003145`

This is the actual call and it should work. Please try with this and merge this PR is it works.

**Review by CrackTheCode016 — APPROVED** [review]

**Pull request merged**